staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -455,58 +455,77 @@ func returnedOpenAPI() *openapi_v2.Document {
 }
 
 func openapiSchemaDeprecatedFakeServer(status int) (*httptest.Server, error) {
```

Comment:
would it make more sense to pass in `t *testing.T` and call `t.Errorf` for these cases to be sure they fail the test

Reply:
To be honest, given the handler has more than one exit, the expectation of cases may be different, we'd better not coupling that.

On the other hand, every case's checkpoint mainly relies on the handler's output, thus if there is something wrong in the handler, they would find it.